global_shortcut.go

```diff
@@ -0,0 +1,123 @@
+		globalShortcutHandler(gs, e.GlobalShortcuts.Accelerator)
+		return
+	})
+	return
+}
+
+// Register Register global shortcuts
+func (gs *GlobalShortcuts) Register(accelerator string, callback callback) (isRegistered bool, err error) {
+
+	if err = gs.ctx.Err(); err != nil {
+		return
+	}
+
+	// Send an event to astilectron to register the global shortcut
+	var event = Event{Name: EventNameGlobalShortcutCmdRegister, TargetID: gs.id, GlobalShortcuts: &EventGlobalShortcuts{Accelerator: accelerator}}
+	result, err := synchronousEvent(gs.ctx, gs, gs.w, event, EventNameGlobalShortcutEventProcessFinished)
+
+	if err != nil {
+		return
+	}
+
+	// If registered successfully, add the callback to the map
+	if result.GlobalShortcuts.IsRegistered {
+		gs.callbacks[accelerator] = &callback
+	}
+
+	isRegistered = result.GlobalShortcuts.IsRegistered
+	return
+}
+
+// IsRegistered Check if a global shortcut is registered
+func (gs *GlobalShortcuts) IsRegistered(accelerator string) (isRegistered bool, err error) {
+
+	if err = gs.ctx.Err(); err != nil {
+		return
+	}
+
+	// Send an event to astilectron to check if global shortcut is registered
+	var event = Event{Name: EventNameGlobalShortcutCmdIsRegistered, TargetID: gs.id, GlobalShortcuts: &EventGlobalShortcuts{Accelerator: accelerator}}
+	result, err := synchronousEvent(gs.ctx, gs, gs.w, event, EventNameGlobalShortcutEventProcessFinished)
+
+	if err != nil {
+		return
+	}
+
+	isRegistered = result.GlobalShortcuts.IsRegistered
+	return
+}
+
+// Unregister Unregister a global shortcut
+func (gs *GlobalShortcuts) Unregister(accelerator string) (err error) {
+
+	if err = gs.ctx.Err(); err != nil {
+		return
+	}
+
+	// Send an event to astilectron to unregister the global shortcut
+	var event = Event{Name: EventNameGlobalShortcutCmdUnregister, TargetID: gs.id, GlobalShortcuts: &EventGlobalShortcuts{Accelerator: accelerator}}
+	_, err = synchronousEvent(gs.ctx, gs, gs.w, event, EventNameGlobalShortcutEventProcessFinished)
+
+	if err != nil {
+		return
+	}
+
+	// No need to find the callback from the map and delete it
+	//  because that event will no longer be triggerred
+	// If the same global shortcut is registered again, the original callback will be replaced with the new one
+
+	return
+}
+
+// UnregisterAll Unregister all global shortcuts
+func (gs *GlobalShortcuts) UnregisterAll() (err error) {
+
+	if err = gs.ctx.Err(); err != nil {
+		return
+	}
+
+	// Send an event to astilectron to unregister all global shortcuts
+	var event = Event{Name: EventNameGlobalShortcutCmdUnregisterAll, TargetID: gs.id}
+	_, err = synchronousEvent(gs.ctx, gs, gs.w, event, EventNameGlobalShortcutEventProcessFinished)
+
+	if err != nil {
+		return
+	}
+
+	gs.callbacks = make(map[string]*callback) // Clear the map
+
+	return
+}
+
+// globalShortcutHandler Handle the GlobalShortcuts event triggered from astilectron
+func globalShortcutHandler(gs *GlobalShortcuts, accelerator string) {
```

Comment:
Could you make this function part of the `GlobalShortcuts` struct instead and rename it to `execCallback`:

```go
func (gs *GlobalShortcuts) execCallback(accelerator string) {
// TODO Lock mutex
// Exec callback
}
```